framework/fp_vectors/Floats.h

```diff
@@ -30,31 +31,46 @@ typedef __fp16 fp16_t;
 #endif
 
 #define FLOAT16_EXPONENT_MASK  0x1fu
+#define BFLOAT16_EXPONENT_MASK 0xffu
 #define FLOAT32_EXPONENT_MASK  0xffu
 #define FLOAT64_EXPONENT_MASK  0x7ffu
 #define FLOAT80_EXPONENT_MASK  0x7fffu
 
 #define FLOAT16_INFINITY_EXPONENT  0x1fu
+#define BFLOAT16_INFINITY_EXPONENT 0xffu
 #define FLOAT32_INFINITY_EXPONENT  0xffu
 #define FLOAT64_INFINITY_EXPONENT  0x7ffu
 #define FLOAT80_INFINITY_EXPONENT  0x7fffu
 
 #define FLOAT16_NAN_EXPONENT  0x1fu
+#define BFLOAT16_NAN_EXPONENT 0xffu
 #define FLOAT32_NAN_EXPONENT  0xffu
 #define FLOAT64_NAN_EXPONENT  0x7ffu
 #define FLOAT80_NAN_EXPONENT  0x7fffu
 
+static_assert(FLOAT16_NAN_EXPONENT == FLOAT16_EXPONENT_MASK, "All Float16::NaNs have all exponent bits set");
+static_assert(FLOAT16_NAN_EXPONENT == FLOAT16_INFINITY_EXPONENT, "Float16::Inf is a special case of Float16::NaN");
+
+static_assert(BFLOAT16_NAN_EXPONENT == BFLOAT16_EXPONENT_MASK, "All BFloat16::NaNs have all exponent bits set");
+static_assert(BFLOAT16_NAN_EXPONENT == BFLOAT16_INFINITY_EXPONENT, "BFloat16::Inf is a special case of BFloat16::NaN");
```

Comment:
You could also compare that they are the same as the FLOAT32 ones because bfloat16 is just a truncated float32